developers/prerequisites/OSGi-tasks.md

```diff
@@ -0,0 +1,198 @@
+
+3. Create a ```CombinedSolarRadio``` that extends ```BaisicRadio``` and implements ```DynamicConsumer``` in ```org.openhab.training.electricity.combinedradio``` it the following way:
+     - the consumer does not need any provider to run;
+     - if one is available, it must set it as current and use it;
+     - if more than one are available, add all to the list with available providers.
+4. What happens when you stop the provider that is not in use? Why does the ```CombinedSolarRadio``` stops and starts? Modify the consumer to handle the removal of the service dynamically - without calling the stop and start methods every time!
+
+     > Hint! Read about service policy in OSGi!
+
+5. Test all the bundles in the OSGi container by starting and stopping the different ```ElectricityProvider``` implementations and track, if the consumers are used as expected.
+
+### V. Events
+
+1. In the ```org.openhab.training.util``` bundle create and register as a service ``` TimeEventSender```, that sends events with topic "org/openhab/training/time" every minute with the current time stamp (hh:mm) (you can use the ```TimeUtilities``` class form chapter [I.](#i-writing-basic-osgi-bundle)).
+
+     > Hint! Install ```org.eclipse.equinox.event``` bundle in order to use *EventAdmin* service for sending events.  
+     > Hint! [ScheduledExecutorService](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html) can be used in this task.
+
+2. Modify the ```org.openhab.training.electricity.tv.TV``` to listen for events with this topic and print an error on the console, when one event is not received.
+
+### VI. Managed Services
+
+1. Implement the [*ManagedService*][ManagedService] interface in the ```org.openhab.training.electricity.tv.TV``` class.
+2. Add a configuration to the ```TV``` class with property "autoSleep" and value a timestamp (hh:mm).
+3. Modify the method that listens for the events with topic "time" from section V.:
+	- when the "autoSleep" property is equal to the "time" topic, display message on the console, that ```TV``` goes to sleep mode and stop listening for events.
+4. Modify the ```Battery``` provider by implementing the *ManagedService* interface. After this modification, it should be possible to change the charge of a battery.
+
+### VII. Console Commands
+
+1. Implement a service in  the ```org.openhab.training.util``` bundle that adds console command, that:
+	- displays all  ```ElectricityProvider``` implementations that are registered with their charge;
+	- displays all consumers, to which providers are they connected (which providers are available) and what is the current provider at the moment;
+
+    > Hint! You might have to register all consumers as services in the OSGi ServiceRegistry.
+
+	- recharges the ```Battery``` with a provided value;
+	- changes the configuration of the ```TV```.
+
+    > Hint! You can add console commands with the help of
```

Comment:
Yes, you are right, I could not find another use of quotes in the rest of the documentation. :)